### ISSUE TITLE
Composer install stuck at writing something into cache and here're something i found

### ISSUE THREAD

**Issue by HuNanZai** [composer/composer#5094]
With the following composer.json:

```json
{
        "require": {
		"php": ">=5.2.4",

		"rmccue/requests": "~1.6.1",
                "hunanzai/log": "~1.1.0",
                "laravelbook/ardent": "~2.4.2"
	},

	"repositories": [
		{
			"type": "vcs",
			"url":  "https://github.com/HuNanZai/log"
		}
	],

	"config": {
		"secure-http": false
	}
}
```

When I run this command:

```
composer install -vvv (please include -vvv!)
```

I get this output:
(the composer just stuck at some step like:)

```
[100.4MB/9.47s] Reading /home/ci123dev/.composer/cache/repo/http---packagist.phpcomposer.com/provider-symfony$config.json from cache

```

Then after google some related questions and trial, i found this:
```
Test A:
1. remove "hunanzai/log": "~1.1.0",
2. run composer install -vvv
3. OK
```
```
Test B:
1. remove "laravelbook/ardent": "~2.4.2"
2. run composer install -vvv
3. OK
````
So i look a little further, this problem may happened in the dependency version analysis "symfony/http-foundation"

```
hunanzai/log: ~1.1.0
        symfony/http-foundation: ~2.0.4

laravelbook/ardent: ~2.4.2
        illuminate/validation: 4.x
                symfony/http-foundation: 2.3.*
```
But i don't know exactly why this would happen and how to prevent from getting this error again.
Ps: the stucking installing step doesn't show anything related to the real problem.

**Comment by curry684** [comment]
The described issue appears similar or even identical to https://github.com/composer/composer/issues/5084. What I found there is that it wasn't just so much infinitely looping as "incredibly freaking slow". The `Reading from` cache log line is misleading as it happens to be the last package being loaded, and nothing is logged before SAT solving is started after that.

**Comment by Seldaek** [comment]
Confusing logging fixed by #5108.

**Comment by edcarlos-92** [comment]
somehow enabling extension=zip in php.ini got this issue resolved
